### PR TITLE
Add commits to /users/stats (for only active users), and create an api call for getting user commits

### PR DESCRIPTION
**Pull request by kmcnellis**

### CODE REVIEW

File: server/api/user/user.controller.js [hunk 1]
```diff
@@ -30,12 +30,40 @@ exports.stats = function(req, res) {
...
+                // }
+                var commitList = [];
+                commits.forEach(function (c){
+                    commitList.push(c.toObject());
+                  }
+                )
+                // console.log(commitList);
+                user.commits = commitList ;
+                count--;
+                console.log(user);
```
Comment (seveibar) [inline server/api/user/user.controller.js]: why is this being logged?

File: server/api/user/user.controller.js [hunk 2]
```diff
@@ -30,12 +30,40 @@ exports.stats = function(req, res) {
...
-    for (var i = 0; i < users.length; i++){
-      data.push(users[i].stats);
+    var count = users.length;
+
+    var getCommits = function(user){
+      Commit.find()
+            .where('author.login').equals(String(user.github.login))
+            .where('date').gt(twoWeeks)
+            .exec(function(err, commits){
+                // if (err) {
```
Comment (seveibar) [inline server/api/user/user.controller.js]: ghost code?

File: server/api/user/user.controller.js [hunk 3]
```diff
@@ -30,12 +30,40 @@ exports.stats = function(req, res) {
...
+                //   return err;
+                // }
+                var commitList = [];
+                commits.forEach(function (c){
+                    commitList.push(c.toObject());
+                  }
+                )
+                // console.log(commitList);
```
Comment (seveibar) [inline server/api/user/user.controller.js]: again, why is there ghost code here?

File: server/api/commit/commit.controller.js [hunk 4]
```diff
@@ -65,6 +65,33 @@ exports.showProjectCommits = function(req, res) {
...
+  }
+  else{
+    twoWeeks.setDate(twoWeeks.getDate()-14);
+  }
+
+  Commit.find()
+        .where('author.login').equals(String(req.params.githubProfile))
+        .where('date').gt(twoWeeks)
+        .exec(function(err, commits){
+          if(err) { return handleError(res, err); }
```
Comment (seveibar) [inline server/api/commit/commit.controller.js]: either use no bracket syntax or split between multiple lines :)

File: server/api/commit/commit.controller.js [hunk 5]
```diff
@@ -65,6 +65,33 @@ exports.showProjectCommits = function(req, res) {
...
+          if(!commits) { return res.send(404); }
```
Comment (seveibar) [inline server/api/commit/commit.controller.js]: see above

File: server/api/commit/commit.controller.js [hunk 6]
```diff
@@ -65,6 +65,35 @@ exports.showProjectCommits = function(req, res) {
...
+  else{
+    prevWeeks.setDate(prevWeeks.getDate()-14);
+  }
+
+  Commit.find()
+        .where('author.login').equals(String(req.params.githubProfile))
+        .where('date').gt(prevWeeks)
+        .exec(function(err, commits){
+          if(err) { return handleError(res, err); }
+          if(!commits) { return res.send(404); }
```
Comment (seveibar) [inline server/api/commit/commit.controller.js]: This is not the correct error type for someone with no commits / a nonexistent person.

File: server/api/user/user.controller.js [hunk 7]
```diff
@@ -30,12 +30,35 @@ exports.stats = function(req, res) {
...
+    var getCommits = function(user){
+      Commit.find()
+            .where('author.login').equals(String(user.github.login))
+            .where('date').gt(twoWeeks)
+            .exec(function(err, commits){
+                var commitList = [];
+                commits.forEach(function (c){
+                    commitList.push(c.toObject());
+                  }
+                )
```
Comment (seveibar) [inline server/api/user/user.controller.js]: parenthesis should not go on next line- not really important but stylistically the parethesis is never really put on a new line

File: server/api/user/user.controller.js [hunk 8]
```diff
@@ -30,12 +30,35 @@ exports.stats = function(req, res) {
...
+                if (count === 0){
+                  res.json(200, data);
+                }
+            });
     }
-    res.json(200, data);
-  });
+
+    for (var i = 0; i < users.length; i++){
+      var u = users[i].stats;
```
Comment (seveibar) [inline server/api/user/user.controller.js]: bad variable name

File: server/api/user/user.controller.js [hunk 9]
```diff
@@ -57,18 +80,13 @@ exports.allStats = function(req, res) {
...
             .exec(function(err, commits){
-                // if (err) {
-                //   return err;
-                // }
                 var commitList = [];
                 commits.forEach(function (c){
                     commitList.push(c.toObject());
                   }
                 )
-                // console.log(commitList);
```
Comment (seveibar) [inline server/api/user/user.controller.js]: nice

File: client/app/profile/profile.controller.js [hunk 10]
```diff
@@ -4,5 +4,10 @@ angular.module('observatory3App')
   .controller('ProfileCtrl', function ($scope, $stateParams, $http) {
       $http.get('/api/users/' + $stateParams.id).success(function(user){
           $scope.user = user;
+          $http.get('/api/commits/user/' + user.githubProfile).success(function(commits){
+              $scope.user.commits = commits;
+              console.log(commits);
```
Comment (seveibar) [inline client/app/profile/profile.controller.js]: why are we logging commits?

File: server/api/commit/commit.controller.js (hunk 6 above)
Comment (kmcnellis) [inline server/api/commit/commit.controller.js]: What would the correct error be?

**Pull request closed without merging**